src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java

```diff
@@ -165,6 +165,22 @@ public CollectionExercise findCollectionExercise(String surveyRef, String exerci
         return linkedSummaries;
     }
 
+    /**
+     * Delete SampleSummary link
+     * @param sampleSummaryId a sample summary uuid
+     * @param collectionExerciseId a collection exercise uuid
+     * @throws CTPException thrown if transition fails
+     */
+    @Override
+    @Transactional
+    public void removeSampleSummaryLink(final UUID sampleSummaryId, final UUID collectionExerciseId)
+            throws CTPException {
+        sampleLinkRepository.deleteBySampleSummaryIdAndCollectionExerciseId(sampleSummaryId, collectionExerciseId);
+
+        transitionScheduleCollectionExerciseToReadyToReview(collectionExerciseId);
```

Comment:
Although it does the right thing in transitioning the state it's a bit weird as it will transition the state based on a CI_ADDED event which is confusing. For constancy we probably need to add another event and do the transition based on that event

Reply:
The events used are CI_SAMPLE_ADDED and CI_SAMPLE_DELETED. But i've changed the logic now to prevent transition getting triggered with a CI_SAMPLE_ADDED event when removing a sample (which would happen if we have multiple sample links for that collection exercise).